omnigibson/robots/manipulation_robot.py

```diff
@@ -1249,18 +1260,31 @@ def _dump_state(self):
         if self.grasping_mode == "physical":
             return state
 
-        # TODO: Include AG_state
-
+        # Include AG_state
+        state["ag_obj_constraint_params"] = self._ag_obj_constraint_params
         return state
 
     def _load_state(self, state):
+        # If there is an existing AG object, remove it
+        self.release_grasp_immediately()
+
         super()._load_state(state=state)
 
         # No additional loading needed if we're using physical grasping
         if self.grasping_mode == "physical":
             return
 
-        # TODO: Include AG_state
+        # Include AG_state
```

Comment:
rubber stamping this for now. i assume dump / loading with AG works properly? (which if it does, is super awesome!!!)

Reply:
debug this - check robot falling scenario - add unit test

Reply:
we'll take care of this in a follow-up.